### PR TITLE
Serialization service improvements

### PR DESCRIPTION
**Pull request by mdumandag**
See the individual commit messages for the explanation.

### CODE REVIEW

File: hazelcast/serialization/data.py [hunk 1]
```diff
@@ -76,17 +77,6 @@ def is_portable(self):
...
 
-    def has_partition_hash(self):
-        """Determines whether this ``Data`` has partition hash or not.
-        
-        Returns:
-            bool: ``True`` if ``Data`` has partition hash, ``False`` otherwise.
-
-        """
-        return self._buffer is not None \
-               and len(self._buffer) >= HEAP_DATA_OVERHEAD \
```
Comment (puzpuzpuz) [inline hazelcast/serialization/data.py:87]: nit: it seems that we don't have this check anymore. Previous implementation was returning `self.hash_code()` when `len(self._buffer) < HEAP_DATA_OVERHEAD`. But I don't think this situation can be faced in the current implementation. WDYT?

**Review by puzpuzpuz — APPROVED** [review]
LGTM

File: hazelcast/serialization/data.py (hunk 1 above)
Comment (mdumandag) [reply hazelcast/serialization/data.py:87]: Yes, I checked that part. We try to get partition hash only when the Data is created from the ObjectDataOutput, which will always be bigger than that

**Comment by mdumandag** [comment]
Thanks for the review Andrey

**Pull request merged**